### DISCUSSION TITLE
column_property in async_scoped_session

### DISCUSSION THREAD

**Discussion by pankajkumar229** [sqlalchemy/sqlalchemy#9117] [Usage Questions] [answered]
It seems column property is not evaluated by async sessions. IS there a way to enable it?

**Comment by pankajkumar229** [comment, marked as answer]
In other words, Is it possible to tell async_scoped_session to load all column_property attributes without explicitly doing a joinedload or defaultload?

**Reply by zzzeek** [reply]
column_property() loads by default when you select the entity, as long as you didnt defer() it

if the object is expired, then it would have to load again, so you'd want to avoid expiration in async

**Reply by pankajkumar229** [reply]
How about when we do a joinedload of another entity? Would that entity's column_property be loaded?

**Reply by pankajkumar229** [reply]
Actually, the column is marked as deferred but accessing it seems to be leading to an error. Should it not load on access?

**Reply by CaselIT** [reply]
No, as documented you can't use lazy load with asyncio.

**Reply by pankajkumar229** [reply]
Thanks!

**Reply by CaselIT** [reply]
If you know that something may not be loaded you can use `run_sync`. Something like `await session.run_sync(lambda s: obj.deferred_prop)`

**Reply by pankajkumar229** [reply]
Thanks! Would running run_sync mean that the event loop is blocked?

**Reply by zzzeek** [reply]
`run_sync()`.  no, because it ultimately uses your database driver for the actual IO which is async.

**Reply by pankajkumar229** [reply]
The name run_sync seems to suggest otherwise. Thanks for the clarification.

**Reply by zzzeek** [reply]
python has made it difficult to come up with terminology here, but this means "run code that does not use the async keyword".  that is "sync" is referring to code that is "synchronous *style*", but it still runs it in the same event loop normally.

**Reply by pankajkumar229** [reply]
Thanks so much for the exemplary help. Much appreciated.

**Reply by CaselIT** [reply]
note that what mike said about not blocking the event loop is valid only for sqlalchemy related calls. You can't place other sync calls (like sleep, request.get, etc) in that callable, since that would block the loop.
